shredstream.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package shredstream;
+
+import "shared.proto";
+
+message Heartbeat {
+  // don't trust IP:PORT from tcp header since it can be tampered over the wire
+  // this also prevents spamming an unwitting destination
+  shared.Socket socket = 1;
+}
+
+// placeholder for future returned values
+message HeartbeatResponse {}
```

Comment:
thoughts on adding a heartbeat cadence field so clients knw how long till they have to heartbeat again? maybe call it `ttl_ms`